src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/JsonContent.Object.cs

```diff
@@ -0,0 +1,66 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#if !NETCOREAPP
+using System.Diagnostics;
+#endif
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Http.Json
+{
+    public sealed partial class JsonContent
+    {
+        private sealed class JsonContentObjectSerializer : JsonContentSerializer
+        {
+            private readonly JsonSerializerOptions? _jsonSerializerOptions;
```

Comment:
Storing `JsonSerializerOptions` is not necessary for this to work. You can always obtain a `JsonTypeInfo` from `JsonSerializerOptions`, roughly:
```C#
internal static JsonContent FromJsonSerializerOptions(Type objectType, object? value, JsonSerializerOptions options)
{
    JsonTypeInfo jsonTypeInfo = options.GetTypeInfo(objectType);
    return new JsonContent(value, jsonTypeInfo); // the same constructor as the one used by JsonTypeInfo overloads.
}
```

Reply:
`JsonContentObjectSerializer ` non-JsonTypeInfo pattern, which requires unreferenced code. This logic was simply moved from the previous implementation of `JsonContent`. It doesn't have access to a `JsonTypeInfo`.